### PR TITLE
Add Coached Conversation Preference Dataset

### PR DESCRIPTION
**Pull request by vineeths96**
Adding [Coached Conversation Preference Dataset](https://research.google/tools/datasets/coached-conversational-preference-elicitation/)

### CODE REVIEW

File: datasets/coached_conv_pref/README.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+
+---
+YAML tags:
```
Comment (lhoestq) [inline datasets/coached_conv_pref/README.md:3]:
```suggestion
---
```

File: datasets/coached_conv_pref/coached_conv_pref.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            {
+                "startIndex": 0,
+                "endIndex": 0,
+                "text": "",
+                "annotations": [
+                    {
+                        "annotationType": "",
+                        "entityType": "",
+                    }
+                ],
```
Comment (lhoestq) [inline datasets/coached_conv_pref/coached_conv_pref.py:157]:
```suggestion
                "annotations": [],
```

you can simply add no annotations in this case no ?

File: datasets/coached_conv_pref/coached_conv_pref.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                                    "text": datasets.Value("string"),
+                                    "annotations": datasets.Sequence(
+                                        {
+                                            "annotationType": datasets.features.ClassLabel(
+                                                names=[
+                                                    "ENTITY_NAME",
+                                                    "ENTITY_PREFERENCE",
+                                                    "ENTITY_DESCRIPTION",
+                                                    "ENTITY_OTHER",
+                                                    "",
```
Comment (lhoestq) [inline datasets/coached_conv_pref/coached_conv_pref.py:99]:
```suggestion
```

File: datasets/coached_conv_pref/coached_conv_pref.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                                                ]
+                                            ),
+                                            "entityType": datasets.features.ClassLabel(
+                                                names=[
+                                                    "MOVIE_GENRE_OR_CATEGORY",
+                                                    "MOVIE_OR_SERIES",
+                                                    "PERSON",
+                                                    "SOMETHING_ELSE",
+                                                    "",
```
Comment (lhoestq) [inline datasets/coached_conv_pref/coached_conv_pref.py:108]:
```suggestion
```
Let's keep only valid classes.
If for some reason one example has a missing label, you can use the `-1` value for this example

**Review by lhoestq — COMMENTED** [review]
Nice thank you !

I left a few comments.
After the change you'll need to regenerate the `dataset_infos.json file`

File: datasets/coached_conv_pref/coached_conv_pref.py (hunk 2 above)
Comment (vineeths96) [reply datasets/coached_conv_pref/coached_conv_pref.py]: Yes, indeed. This looks much cleaner!

File: datasets/coached_conv_pref/coached_conv_pref.py (hunk 4 above)
Comment (vineeths96) [reply datasets/coached_conv_pref/coached_conv_pref.py]: @lhoestq I have made all the changes you proposed. Please take a look at it and let me know of any improvements.

**Review by lhoestq — APPROVED** [review]
Thanks ! looks good to me :)

**Pull request merged**